src/Model.js

```diff
@@ -545,10 +545,10 @@ export class Model {
         try {
             await this.table.transact('write', params.transaction, params)
         } catch (err) {
-            if (err.message.indexOf('ConditionalCheckFailed') >= 0) {
+            if (err instanceof OneTableError && err.code === 'TransactionCanceledException' && err.context.err.CancellationReasons.map(r => r.Code).includes('ConditionalCheckFailed')) {
```

Comment:
Does this need to be more blunt to work with dynamodb v2? ie. not checking CancellationReasons

Reply:
Sorry, I don't have immediate understanding of what the V2 equivalent would be.

I do seem to remember that V2 did not include such transaction failure errors. The GO SDK did and there was a message somewhere that said V2 would not support it in Javascript.